portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +27,44 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  private List<String> comments;
+
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Erys Dennis!</h1>");
+    response.setContentType("application/json;");
+    String json = new Gson().toJson(comments);
+    response.getWriter().println(json);
+  }
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String comment = getParameter(request, "text-input", "");
+    if (comment != ""){
+      comments.add(comment);
+      response.setContentType("text/html;");
+      response.sendRedirect("/index.html");
```

Comment:
You'll be sending 2 redirects in this case.  I don't think you need either of these

Reply:
Now you're sending no redirects.  You should keep the 1 at the end